yt_dlp/extractor/tunein.py

```diff
@@ -1,244 +1,332 @@
     _TESTS = [{
-        'url': 'https://tunein.com/podcasts/Technology-Podcasts/Artificial-Intelligence-p1153019',
+        'url': 'https://tunein.com/podcasts/Technology-Podcasts/Artificial-Intelligence-p1153019/',
         'info_dict': {
             'id': 'p1153019',
             'title': 'Lex Fridman Podcast',
-            'description': 'md5:bedc4e5f1c94f7dec6e4317b5654b00d',
         },
         'playlist_mincount': 200,
-    }, {
-        'url': 'https://tunein.com/embed/player/p191660/',
-        'only_matching': True,
     }, {
         'url': 'https://tunein.com/podcasts/World-News/BBC-News-p14/',
         'info_dict': {
             'id': 'p14',
             'title': 'BBC News',
-            'description': 'md5:30b9622bcc4bd101d4acd6f38f284aed',
         },
-        'playlist_mincount': 36,
+        'playlist_mincount': 35,
     }]
 
-    _PAGE_SIZE = 30
-
-    def _real_extract(self, url):
-        podcast_id = self._match_id(url)
-
-        webpage = self._download_webpage(url, podcast_id, fatal=False)
-        metadata = self._extract_metadata(webpage, podcast_id)
-
-        def page_func(page_num):
-            api_response = self._download_json(
-                f'https://api.tunein.com/profiles/{podcast_id}/contents', podcast_id,
-                note=f'Downloading page {page_num + 1}', query={
+    def _entries(self, podcast_id):
+        for page in itertools.count():
+            contents = self._call_api(
+                podcast_id, 'contents',
+                f'Downloading page {page + 1}', query={
                     'filter': 't:free',
-                    'offset': page_num * self._PAGE_SIZE,
                     'limit': self._PAGE_SIZE,
+                    'offset': page * self._PAGE_SIZE,
                 })
 
-            return [
-                self.url_result(
-                    f'https://tunein.com/podcasts/{podcast_id}?topicId={episode["GuideId"][1:]}',
-                    TuneInPodcastEpisodeIE, title=episode.get('Title'))
-                for episode in api_response['Items']]
+            yield from traverse_obj(contents, (
+                'Items', ..., 'GuideId', {str}, filter, all, filter))
+
+            if not traverse_obj(contents, ('Paging', 'Next', {url_or_none})):
+                break
+
+    def _real_extract(self, url):
+        podcast_id = self._match_id(url)
 
-        entries = OnDemandPagedList(page_func, self._PAGE_SIZE)
-        return self.playlist_result(
-            entries, playlist_id=podcast_id, title=traverse_obj(metadata, ('profiles', podcast_id, 'title')),
-            description=traverse_obj(metadata, ('profiles', podcast_id, 'description')))
+        return self.playlist_from_matches(
+            self._entries(podcast_id), podcast_id,
+            traverse_obj(self._call_api(podcast_id), ('Item', 'Title', {str})),
+            getter=lambda x: update_url_query(url, {'topicId': x[1:]}))
```

Comment:
it's always best to use a `PagedList` if we can; this makes options like `--playlist-items` work much more efficiently